drivers/user_permission_report/app/controllers/user_permission_report/warehouse_reports/reports_controller.rb

```diff
@@ -6,37 +6,56 @@
 
 module UserPermissionReport::WarehouseReports
   class ReportsController < ApplicationController
-
     before_action :set_group_associations
 
     def index
-      if params[:q].present?
-        @users = User.active.
-          text_search(params[:q]).
-          order(:last_name, :first_name).
-          includes(:roles, access_groups: @group_associations).
-          page(params[:page]).per(25)
-      else
-        @users = User.active.order(:last_name, :first_name).includes(:roles, access_groups: @group_associations).page(params[:page]).per(25)
+      respond_to do |format|
+        format.html do
+          if params[:q].present?
+            @users = User.active.
+              text_search(params[:q]).
+              order(:last_name, :first_name).
+              includes(:roles, access_groups: @group_associations).
+              page(params[:page]).per(25)
+          else
+            @users = User.active.order(:last_name, :first_name).includes(:roles, access_groups: @group_associations).page(params[:page]).per(25)
+          end
+        end
+        format.xlsx do
+          @users = User.active.
+            order(:last_name, :first_name).
+            includes(:roles, access_groups: @group_associations)
+          date = Date.current.strftime('%Y-%m-%d')
+          filename = "user-permissions-#{date}.xlsx"
+          headers['Content-Disposition'] = "attachment; filename=#{filename}"
+        end
       end
     end
 
+    def self.all_associations
+      {
+        data_sources: GrdaWarehouse::DataSource,
+        organizations: GrdaWarehouse::Hud::Organization,
+        projects: GrdaWarehouse::Hud::Project,
+        reports: GrdaWarehouse::WarehouseReports::ReportDefinition,
+        cohorts: GrdaWarehouse::Cohort,
+        project_groups: GrdaWarehouse::ProjectGroup,
+      }
```

Comment:
Probably want `.freeze` on the end of this.

Reply:
✅ We resolved this a different way by moving group_associations to the user model.